src/ng/directive/ngOptions.js

```diff
@@ -303,6 +303,13 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         values = values || [];
 
         Object.keys(values).forEach(function getWatchable(key) {
+          if (isString(values[key]) &&
```

Comment:
Also as I mention further down, it makes no sense to care about array values, period